### PR TITLE
fix: use ip address in forgot password rate limitation

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

fix for the comments from https://github.com/calcom/cal.com/pull/7700#discussion_r1140716657

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: apps/web/pages/api/auth/forgot-password.ts [hunk 1]
```diff
@@ -15,13 +15,21 @@ const limiter = rateLimit({
...
     .email()
     .transform((val) => val.toLowerCase())
     .parse(req.body?.email);
 
-  const { isRateLimited } = limiter.check(10, email); // 10 requests per minute
+  let ip = req.headers["x-real-ip"] as string;
+
+  const forwardedFor = req.headers["x-forwarded-for"] as string;
+  if (!ip && forwardedFor) {
+    ip = forwardedFor?.split(",").at(0) ?? email;
```
Comment (G3root) [inline apps/web/pages/api/auth/forgot-password.ts:29]: FYI https://vercel.com/docs/concepts/edge-network/headers#x-forwarded-for there's warning that IP-address won't be forwarded if Vercel is behind a proxy.

**Review by zomars — DISMISSED** [review]
How can we test this?

**Comment by G3root** [comment]
@zomars  here's minimal reproduction for this https://ip-ratelimit-test.vercel.app/. requests are rate limited 5 per minute.

source code: https://github.com/G3root/ip-ratelimit-test

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
Screenshot 2023-04-11 at 1 55 23 AM

I am getting 500 internal server error instead of 429 too many Requests

**Comment by G3root** [comment]
@Udit-takkar pushed a fix for that

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**